docs/source/known-issues.rst

```diff
@@ -4,6 +4,20 @@ Known issues
 .. role:: cppkokkos(code)
     :language: cppkokkos
 
+CUDA
+====
+
+- When using a legacy NVIDIA gpu, Kokkos' initialization can report errors like:
```

Comment:
Do we have a good idea of what GPUs have this issue?  (If so it might be good to say "(e.g. pre Pascal architectures)" or whatnot.
Is it sensitive to the CUDA version?  (Is there something we could do on the Kokkos side?  The note is valuable regardless.)

Reply:
I'll try to see. From what I remember, it was not easy to test on the Kokkos side, it depends both on the GPU and on how the system is configured (perhaps the issue with the other MPI backends is related to this).

The one report on slack was with sm_35 and cuda 11.8.  I may dive a little bit deeper in Nvidia documentation to see if from Pascal it is automatically safe or not.

Reply:
According to https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__MEMORY__POOLS.html , what architectures are supported is not very clear.

However, we could do a better job at checking availability in Kokkos:
> Supported Platforms

> Whether or not a device supports the integrated stream ordered memory allocator may be queried by calling [cudaDeviceGetAttribute()](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__DEVICE.html#group__CUDART__DEVICE_1gb22e8256592b836df9a9cc36c9db7151) with the device attribute [cudaDevAttrMemoryPoolsSupported](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html#group__CUDART__TYPES_1gg49e2f8c2c0bd6fe264f2fc970912e5cd0876515d3b274cb0b40f8a7d913fb1aa).

Reply:
I think so.

Reply:
See https://github.com/kokkos/kokkos/issues/7044.